Content.Server/CombatMode/CombatModeSystem.cs

```diff
@@ -89,8 +102,65 @@ private void OnEntityActionPerform(EntityUid uid, SharedCombatModeComponent comp
             SoundSystem.Play(component.DisarmSuccessSound.GetSound(), filterAll, args.Performer, AudioHelpers.WithVariation(0.025f));
             _adminLogger.Add(LogType.DisarmedAction, $"{ToPrettyString(args.Performer):user} used disarm on {ToPrettyString(args.Target):target}");
 
-            var eventArgs = new DisarmedEvent() { Target = args.Target, Source = args.Performer, PushProbability = component.DisarmPushChance };
+            var eventArgs = new DisarmedEvent() { Target = args.Target, Source = args.Performer, PushProbability = chance };
             RaiseLocalEvent(args.Target, eventArgs);
         }
+
+
+        private float CalculateDisarmChance(EntityUid disarmer, EntityUid disarmed, EntityUid? inTargetHand, SharedCombatModeComponent disarmerComp)
+        {
+            Logger.Error("inTargetHand is: " + inTargetHand);
+
+            float healthMod = 0;
+            if (!TryComp<DamageableComponent>(disarmer, out var disarmerDamage) || !TryComp<DamageableComponent>(disarmed, out var disarmedDamage))
+            {
+               healthMod = 0; // If some of them can't take damage, no effect from this.
```

Comment:
you're already setting it to 0 before this, just invert this and put the else code here